lambdas/services/document_service.py

```diff
@@ -89,12 +104,12 @@ def delete_documents(self, table_name: str, documents: list[DocumentReference]):
 
         logger.info(f"Deleting items in table: {table_name}")
 
-        for doc in documents:
+        for reference in document_references:
             self.s3_service.create_object_tag(
-                file_key=doc.get_file_key(),
-                s3_bucket_name=doc.get_file_bucket(),
-                tag_key=str(S3LifecycleTags.SOFT_DELETE_KEY.value),
-                tag_value=str(S3LifecycleTags.SOFT_DELETE_VAL.value),
+                file_key=reference.get_file_key(),
+                s3_bucket_name=reference.get_file_bucket(),
+                tag_key=str(S3LifecycleTags.SOFT_DELETE.value),
+                tag_value=str(S3LifecycleTags.ENABLE_TAG.value),
             )
 
-            self.update_item(table_name, doc.id, updated_fields=update_fields)
+            self.update_item(table_name, reference.id, updated_fields=update_fields)
```

Comment:
It appears to me that this part of code will put a soft_delete tag, even if type_of_delete is death_delete. Is that something we could have a look ?

Reply:
Good spot will change that now